packages/sdk-android/src/deviceManager.ts

```diff
@@ -443,11 +443,48 @@ export const connectToWifiDevice = async (target: string) => {
     }
 
     const deviceResponse = await execCLI(CLI_ANDROID_ADB, connect_str);
+
     if (deviceResponse.includes('connected')) return true;
-    logError(`Failed to ${connect_str}`, { skipAnalytics: true });
+
+    if (deviceResponse.includes('Connection refused')) {
+        logError(`Failed to ${connect_str}. Connection refused. Make sure to that ip and port are correct.`, {
+            skipAnalytics: true,
+        });
+        return false;
+    }
+
+    if (!deviceResponse.includes('Connection refused')) {
```

Comment:
@Marius456 this if statement is not needed as you already checking above
`if (deviceResponse.includes('Connection refused')) {` with return clause.